src/tests/Interop/Swift/README.md

```diff
@@ -0,0 +1,96 @@
+ - Mutating method on value types: pointer to the value (i.e. value is passed indirectly)
+ - Non-mutating methods on value types: self may fit in one or more registers, else passed indirectly
+ - Thick closures, i.e. closures requiring a context: the closure context
+
+Error handling is also handled through registers, so the caller needs to check for errors and throw if necessary. Similarly, the async context is handled through a register which contains a pointer to an object with information about the current state of the asynchronous operation. More details available at https://github.com/apple/swift/blob/main/docs/ABI/CallConvSummary.rst.
+
+### Name mangling
+
+Swift uses mangling for generating unique names in a binary. This process can change in different major versions (i.e. Swift 4 vs Swift 5). The Swift compiler puts mangled names in binary images to encode type references for runtime instantiation and reflection. In a binary, these mangled names may contain pointers to runtime data structures to represent locally-defined types more efficiently.
+
+Be aware of the edge cases that can't be mapped 1:1 in C#. Consider the following example in Swift:
+```swift
+public static func getValue() -> Double {
+    return 5.0
+}
+
+public static func getValue() -> Int {
+    return 5
+}
+```
+
+The Swift compiler generates the following mangled names for the functions:
+```
+_$s11MathLibraryAAC8getValueSdyFZ ---> static MathLibrary.MathLibrary.getValue() -> Swift.Double
+_$s11MathLibraryAAC8getValueSiyFZ ---> static MathLibrary.MathLibrary.getValue() -> Swift.Int
+```
+
+In such case, generating entry points automatically from the C# would be problematic because both methods have the same name and parameter list, but different return types. According to the C# documentation, a return type of a method is not part of the signature of the method for the purposes of method overloading.
+
+https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/methods#method-signatures
+
+The Binding Tools for Swift resolves that by reading the native library, extracting the public symbols, and demangling them. On the runtime level the user is supposed to provide the mangled name as the entry point.
+
+## Goals
+
+The goal of this experiment is to explore the possibilities and limitations of direct P/Invoke interop with Swift. It should implement runtime mechanisms for handling Swift ABI differences. In the first iteration of the experiment, our focus will be on the self context and error handling calling conventions with blittable types targeting MacCatalyst. After that, we plan to expand support for non-blittable types using binding wrappers and to include other Apple targets.
+
+We can choose to first support Mono AOT either with or without LLVM according to the initial analysis.
+
+## Tasks
+
+Progress on the implementation can be tracked at https://github.com/dotnet/runtime/issues/93631.
+
+## Local build and testing
```

Comment:
This is a duplicate of data that should be under https://github.com/dotnet/runtime/tree/main/docs/workflow/testing

Reply:
Yes, this will be removed.